pkg/operator/ceph/cluster/watcher_test.go

```diff
@@ -32,6 +32,43 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+func TestOnK8sNode(t *testing.T) {
+	service := &corev1.Service{}
+	ns := "rook-ceph"
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(cephv1.SchemeGroupVersion, &cephv1.CephCluster{})
+
+	cephCluster := &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+		Status: cephv1.ClusterStatus{
+			Phase: "",
+		},
+	}
+
+	object := []runtime.Object{
+		cephCluster,
+	}
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(object...).Build()
+
+	clientCluster := newClientCluster(cl, ns, &clusterd.Context{})
+
+	// Dummy object
+	b := clientCluster.onK8sNode(service)
+	assert.False(t, b)
+
+	node := &corev1.Node{}
+
+	// node not ready for reconcile
+	b = clientCluster.onK8sNode(node)
+	assert.False(t, b)
```

Comment:
Both `true` and `false` conditions should be tested. To test the `true` conditions, output of `getOSDonHost` can be mocked.[ For example](https://github.com/rook/rook/blob/fcf634e3b7808e4338e7fd261c047aae4793e99b/pkg/daemon/ceph/client/crush_test.go#L249-L256)

Reply:
Done! :)